Library/Homebrew/dev-cmd/cat.rb

```diff
@@ -31,8 +32,16 @@ def cat
 
     cd HOMEBREW_REPOSITORY
     pager = if Homebrew::EnvConfig.bat?
+      unless Formula["bat"].any_version_installed?
+        # The user might want to capture the output of `brew cat ...`
+        # Redirect stdout to stderr
+        redirect_stdout($stderr) do
+          ohai "Installing `bat` for displaying <formula>/<cask> source..."
```

Comment:
Nice work! I think `update-report.rb` has some usage that would benefit from using this. I also wonder if we want a general `install_formula!` or something method that does this `any_version_installed` (or `latest_version_installed` check) and prints these messages to `stderr`? There's definitely a few other places that could benefit from that?

Reply:
@MikeMcQuaid I think I can create another PR for adding method `Formula::ensure_installed!` and update places using `(any|latest)_version_installed?` + `safe_system HOMEBREW_BREW_FILE, ...` to that new function.

Reply:
>   451          ohai_stdout_or_stderr "Installing #{new_name}..."

Yeh, this is the function I meant that could be removed and replaced 👍🏻 


> I think I can create another PR for adding method `Formula::ensure_installed!` and update places using `(any|latest)_version_installed?` + `safe_system HOMEBREW_BREW_FILE, ...` to that new function.

That sounds great. I think it'd be good to put in `Utils` rather than `Formula` though as I think it could conditionally load `formula.rb`. There will need to be some conditional parameters e.g. on whether you need `any` or `latest` version and whether to satisfy with a `which`.